wintasee/MemoryManager/MemoryManager.cpp

```diff
@@ -0,0 +1,683 @@
+/*
+ * (c) 2015- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#include <Windows.h>
+
+#include <atomic>
+#include <list>
+
+#include "MemoryManager.h"
+#include <print.h>
+#include <Utils.h>
+
+namespace MemoryManagerInternal
+{
+    enum MemoryBlockState : UINT
+    {
+        MEMORY_BLOCK_USED = 0x00000001,
+        MEMORY_BLOCK_FREE = 0x00000002,
+    };
+
+    /*
+     * 'head' is a pointer to the variable containing the head pointer.
+     * It must be kept valid at all times.
+     * This saves us from having to keep track of which object the allocation belongs to when
+     * allocating new sub-blocks.
+     */
+    struct AddressLinkedList
+    {
+        AddressLinkedList* prev;
+        AddressLinkedList* next;
+        AddressLinkedList** head;
+        LPVOID address;
+        UINT bytes;
+        UINT flags;
+    };
+
+    struct MemoryBlockDescription :
+        public AddressLinkedList
+    {
+    };
+
+    struct MemoryObjectDescription :
+        public AddressLinkedList
+    {
+        HANDLE object;
+        MemoryBlockDescription* blocks;
+    };
+
+    bool memory_manager_inited = false;
+
+    LPVOID minimum_allowed_address = nullptr;
+    LPVOID maximum_allowed_address = nullptr;
+    DWORD allocation_granularity = 0;
+    UINT size_of_mbd = sizeof(MemoryBlockDescription) + (8 - (sizeof(MemoryBlockDescription) % 8));

```

Comment:
Was this meant to be `UINT size_of_mbd = sizeof(MemoryBlockDescription) + (8 - (sizeof(MemoryBlockDescription) % 8)) % 8;`? (extra `% 8` in the end)